api_spec.yaml

```diff
@@ -3246,9 +3246,451 @@ paths:
+    post:
+      tags:
+        - Multitenancy Recipe
+      description: |
+        Delete ThirdParty Provider config for a tenant.
+
+        If `tenantId` is not provided, `defaultTenantId` will be used.
+      operationId: deleteTenantConfigPOST
+      parameters:
+        - $ref: '#/components/parameters/multitenancyRid'
+        - $ref: '#/components/parameters/api-key'
+        - $ref: '#/components/parameters/cdi-version'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - thirdPartyId
+              properties:
+                thirdPartyId:
+                  $ref: '#/components/schemas/thirdPartyId'
+                tenantId:
+                  $ref: '#/components/schemas/tenantId'
+
+      responses:
+        200:
+          description: Redirect URL
+          content:
+            application/json:
+              schema:
+                oneOf:
+                - type: object
+                  properties:
+                    status:
+                      $ref: '#/components/schemas/statusOK'
+                    didConfigExist:
+                      type: boolean
+                - type: object
+                  required:
+                    - status
+                  properties:
+                    status:
+                      $ref: '#/components/schemas/statusTenantDoesNotExistError'
+        '400':
+          $ref: '#/components/responses/400'
+        
+        '401':
+          $ref: '#/components/responses/401'
+        
+        '402':
+          $ref: '#/components/responses/402'
+
+        '404':
+          $ref: '#/components/responses/404'
+        
+        '500':
+          $ref: '#/components/responses/500'
+
+  /recipe/multitenancy/config/thirdparty/provider/list:
+    get:
+      tags:
+        - Multitenancy Recipe
+      description: |
+        Returns the list of thirdparty providers for the given input
+
+        - if only `thirdPartyId` is specified: returns provider configs across all tenants for the `thirdPartyId`
+        - if only `tenantId` is specified: returns all thirdparty provider configs for that `tenantId`
+        - if both (`thirdPartyId`  & `tenantId`) are not specified: returns all thirdparty provider configs for the default tenant (`defaultTenantId`)
+        - if both (`thirdPartyId`  & `tenantId`) are specified: either returns an empty list if no config for that `thirdPartyId` & `tenantId`, or returns one element with the config for that `thirdPartyId` & `tenantId`
+      operationId: listConfigsGET
+      parameters:
+        - $ref: '#/components/parameters/multitenancyRid'
+        - $ref: '#/components/parameters/api-key'
+        - $ref: '#/components/parameters/cdi-version'
+        - name: thirdPartyId
+          in: query
+          required: false
+          schema:
+            $ref: '#/components/schemas/thirdPartyId'
+        - name: tenantId
```

Comment:
No need to have tenantId as input to this API - redo this API

Reply:
Done